### DISCUSSION TITLE
Play 2.8.19

### DISCUSSION THREAD

**Discussion by mkurz** [playframework/playframework#11606] [Announcements]
The Play Team is happy to announce the release of Play 2.8.19 :partying_face: 

## ping/pong support for WebSockets :ping_pong:

[Finally](https://github.com/playframework/playframework/pull/11521)! Check out the documentation:
* [WebSockets Play Scala](https://www.playframework.com/documentation/2.8.19/ScalaWebSockets#Configuring-keep-alive-Frames)
* [WebSockets Play Java](https://www.playframework.com/documentation/2.8.19/JavaWebSockets#Configuring-keep-alive-Frames)

## Using Netty, WebSockets time out now :hourglass_flowing_sand: 
[Another fix](https://github.com/playframework/playframework/pull/11420) for WebSockets ships with this release as well:
If you are using the Netty backend the `play.server.http[s].idleTimeout` setting will now be honored for WebSocket connections. Until now, when using Netty, a WebSocket connection never timed out. That might even was desirable for some use cases, but now that we have ping/pong support you have to make use of that to keep WebSocket connections open. That is the correct way of doing things and not closing connections after an idle timeout was actually a bug. The akka-http backend was always working correctly and didn't need to be fixed.

## Removed the shutdown hook from the default logback config :hook: 

If you have

```xml
<shutdownHook class="ch.qos.logback.core.hook.DelayingShutdownHook"/>
```
in your logback config, you should remove that line. Play handles the shutdown of the logger context, the line shown is not necassary anymore since at least Play 2.7 and is a leftover that should have been removed a while ago. More details can be found in the [according pull request](https://github.com/playframework/playframework/pull/11532).

## Correctly encode `Content-Disposition: form-data; name="..."; filename="..."` :capital_abcd: 

When Play renders a request body containing multipart/form data it will now encode the name and the filename fields according the ["WHATWG HTML living standard" section 4.10.21.8](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#multipart-form-data). `curl`, Firefox, Chrome and other libraries like Python's urllib3 follow the same approach. Until now, Play didn't encode those fields which could result in security implications. Just to be clear, this is not about receiving and parsing multipart/form data from a client, but when sending multipart/form data e.g. via ws or when using the `RequestBuilder` to build a request for testing purposes. Details can be found in pull request #11571.

## A few words on Play's compatibility with sbt 1.8 :wrench: 

Now that sbt 1.8 got released there have been reports that it isn't working out of the box with Play, caused by a version conflict regarding scala-xml, see #11522. Unfortunately, right now, we don't have a fix for that yet. Even if we bump scala-xml to version 2.x in Play and all its dependencies, we can't currently make sbt 1.8 work with Play because of another problem described in #11527.
The good news is, that isn't a showstopper if you still want to upgrade to sbt 1.8. You can do that right now by putting
```scala
ThisBuild / libraryDependencySchemes += "org.scala-lang.modules" %% "scala-xml" % VersionScheme.Always
```
in `project/plugins.sbt`. That is possible because scala-xml 1.x and 2.x are compatible anyway. However, like described in #11527 you will not be able to use `PlayNonBlockingInteractionMode` with sbt 1.8 then (which probably not many people do anyway)

## Further roadmap :world_map: 

...news coming soon...

# Merged pull requests

Following pull requests got merged for this release:

- #11604 [2.8.x] Upgrade netty + netty reactive streams + fix backports by @mkurz
- #11603 [2.8.x] Escape Content-Disposition params according to WHATWG HTML living standard (backport #11571) by @mkurz
- #11602 [2.8.x] Add exceptionOverrideClassName Hikari config setting (backport #11601) by @benwaffle
- #11598 [2.8.x] Fix the example code to DiscardingCookie for Results#discardingCookies (backport #11597) by @tsuyoshizawa
- #11593 Update PlayApplicationOverview.md by @SenthilNayagan
- #11528 [2.8.x] Upgrade Scalafmt to latest release of 2.4.x serias (backport #11524) by @ihostage
- #11567 [2.8.x] Added play-hmac-signatures (backport #11559) by @phelps-sg
- #11560 [2.8.x] Updating documentation - ConfiguringHttps.md by @szymond
- #11552 Update SettingsJDBC.md by @bmontuelle
- #11532 [2.8.x] Do not use/recommend shutdownHook in Logback, Play handles that by @mkurz
- #11519 [2.8.x] Log websocket communication problems + handle special Netty case by @mkurz
- #11521 [2.8.x] Add WebSocket ping/pong support by @mkurz
- #11420 [2.8.x] Netty: Websockets should respect play.server.http.idleTimeout by @mkurz
- #11518 [2.8.x] Use environment file instead `set-output` by @ihostage

For more details see the [full list of changes](https://github.com/playframework/playframework/compare/2.8.18...2.8.19/) and the [2.8.19 milestone](https://github.com/playframework/playframework/issues?page=1&q=milestone%3A%222.8.19%22).

## :heart: Thanks to our premium sponsors!

<div align="center">
  <a href="https://informaticon.com/"><img src="https://www.playframework.com/assets/images/home/sponsors/49a3d5258c8e8b1daf5481eb9e00b898-informaticon-logo-black.png" width="250"></a>
  <a href="https://cedarlakeventures.com/"><img src="https://www.playframework.com/assets/images/home/sponsors/bec2b526c9ce52c051f9089a10044867-cedar-lake-ventures.png" width="250"></a>
  <a href="https://iterable.com/"><img src="https://www.playframework.com/assets/images/home/sponsors/61ddb4c3665b621e6672181f97196748-iterable.png" width="250"></a>
  <a href="https://pronto.net/"><img src="https://www.playframework.com/assets/images/home/sponsors/c77b1d664f10a1c9cb19b97c6d8bd204-pronto-software.png" width="250"></a>
  <a href="https://civiform.us/"><img src="https://www.playframework.com/assets/images/home/sponsors/cb047b3782866c962c4d6a35b056b809-civiform.png" width="250"></a>
  <a href="https://theguardian.com/"><img src="https://www.playframework.com/assets/images/home/sponsors/b15eb0f249dbc45089872e268d8ea5ad-the_guardian.png" width="250"></a>
</div>

If you find this OSS project useful for work, please consider asking your company to support it by <a href="https://www.playframework.com/sponsors">becoming a sponsor</a>.
You can also individually sponsor the project by <a href="https://www.playframework.com/sponsors">becoming a backer</a>.

<div align="center">
  <a href="https://opencollective.com/playframework" target="_blank">
    <img src="https://opencollective.com/playframework/donate/button@2x.png?color=blue" width="250" />
  </a>
</div>

## :bow: Thanks to our contributors

Finally, thanks to the community for their help with detailed bug reports, discussions about new features and pull request reviews. This project is only possible due to the help we had from amazing contributors.
Special thanks to all code contributors who helped with this particular release (they are listed below)!

<hr /><em>This discussion was created from the release <a href='https://github.com/playframework/playframework/releases/tag/2.8.19'>Play 2.8.19</a>.</em>

**Comment by cyrilfr** [comment]
Thanks for your work @mkurz :)
